lib/components/ConnectionParamsControl.jsx

```diff
@@ -0,0 +1,242 @@
+ *    Semiconductor ASA integrated circuit in a product or a software update for
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/* eslint react/forbid-prop-types: off */
+/* eslint jsx-a11y/label-has-for: off */
+/* eslint jsx-a11y/label-has-associated-control: off */
+/* eslint-disable react/destructuring-assignment */
+
+'use strict';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import TextInput from './input/TextInput';
+
+const SUCCESS = 'success';
+const ERROR = 'error';
+
+class ConnectionParamsControl extends React.PureComponent {
+    state = {
+        minConnectionInterval: this.props.connectionParameters
+            .minConnectionInterval,
+        maxConnectionInterval: this.props.connectionParameters
+            .maxConnectionInterval,
+        slaveLatency: this.props.connectionParameters.slaveLatency,
+        connectionSupervisionTimeout: this.props.connectionParameters
+            .connectionSupervisionTimeout,
+    };
+
+    validateMinConInterval() {
+        const { minConnectionInterval, maxConnectionInterval } = this.state;
+        if (
+            minConnectionInterval < 7.5 ||
+            minConnectionInterval > 4000 ||
+            minConnectionInterval > maxConnectionInterval
+        ) {
+            return ERROR;
+        }
+        return SUCCESS;
+    }
+
+    validateMaxConInterval() {
```

Comment:
Consider extracting the duplicated parts of the validation functions into a new function. For example:

```js
const validationWrapper = (predicate) => (predicate ? ERROR : SUCCESS);

// Now validation functions becomes this instead:

validateMaxConInterval() {
        const { minConnectionInterval, maxConnectionInterval } = this.state;
        return validationWrapper(
            maxConnectionInterval < 7.5 ||
                maxConnectionInterval > 4000 ||
                minConnectionInterval > maxConnectionInterval
        );
    }

```

I would also consider using full names for the functions, e.g. `validateMaxConnectionInterval` instead of `validateMaxConInterval`. It's not much longer and might in some cases reduce potential confusion.